R/readIEA_EnergyEfficiencyIndicators.R

```diff
@@ -0,0 +1,26 @@
+#' Read-in data from IEA End Uses and Efficiency Indicators Database
+#'
+#' @author Falk Benke
+#' @importFrom data.table fread :=
+#' @importFrom dplyr %>% filter mutate distinct
+readIEA_EnergyEfficiencyIndicators <- function() { #nolint object_name_linter
+
+  data <- NULL
+
+  for (domain in c("INDUSTRY", "TRANSPORT", "RESIDENTIAL", "SERVICES")) {
+    data <- rbind(data, fread(
+      file = file.path("2023", paste0("IEA - EEI ", domain, ".TXT")),
+      col.names = c("CARBON_INDICATOR", "END_USE", "TIME", "COUNTRY", "VALUE"),
```

Comment:
```suggestion
      col.names = c("ITEM", "ENDUSE", "TIME", "COUNTRY", "VALUE"),
```
I suggest different dimension names. This is tricky, as IEA mixes very different types of data:
- `col.names[1]`: Indicators, Activity, Products --> ITEM
- `col.names[2]`: End use, _blank for activity_ --> ENDUSE